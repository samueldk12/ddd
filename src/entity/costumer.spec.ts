import Address from "./address";
import Costumer from "./costumer";

describe("Costumer unit tests", () => {
    it("should trow error when id is empty",()=>{
        expect(()=>{
            let constumer = new Costumer('','John Doe');
        }).toThrowErrorMatchingSnapshot("Id is required")
    })

    it("should trow error when name is empty",()=>{
        expect(()=>{
            let constumer = new Costumer('123','');
        }).toThrowErrorMatchingSnapshot("Name is required")
    })

    it("should change costumer name",()=>{
        let costumer = new Costumer('123','John Doe');
        costumer.changeName('Jane');
        expect(costumer.name).toBe('Jane');
    })

    it("should activate client",()=>{
        let costumer = new Costumer('123', 'John Doe');
        const address = new Address("1", "123", 13, "213131","BH","")
        costumer._address = address;

        costumer.activate();
        expect(costumer.isActive()).toBe(true);
    })
});