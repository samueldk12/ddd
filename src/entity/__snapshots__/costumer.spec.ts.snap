// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Costumer unit tests should trow error when id is empty: Id is required 1`] = `"Costumer invalid"`;

exports[`Costumer unit tests should trow error when name is empty: Name is required 1`] = `"Costumer invalid"`;
